monkey/monkey_island/cc/services/analytics.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+import requests
+
+from monkey_island.cc import Version
+from monkey_island.cc.deployment import Deployment
+
+logger = logging.getLogger(__name__)
+
+ANALYTICS_URL = (
+    "https://m15mjynko3.execute-api.us-east-1.amazonaws.com/default?version={"
+    "version}&deployment={deployment}"
+)
+
+
+class Analytics:
```

Comment:
This doesn't make sense as a class. It doesn't maintain any state and nothing uses any object instances of this class. the constructor calls `_send_analytics()` to do some work and how it actually gets called is triggered when it gets added to the DI container. Very mysterious.

This can just be a function and you pass the Version and Deployment into it. In the long run, the version and deployment will be constructed at an earlier phase of the boot process.

Reply:
I agree, the "when" part is not evident. After taking a second look at the DI container I found a way to use it directly instead of creating a class